### ISSUE TITLE
AppDomain.AssemblyResolve sometimes has null RequestingAssembly

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#24483]
There is a report of AppDomain.AssemblyResolve setting RequestingAssembly to null while running 3.0.100-preview5-011235. 

``` 
*** NULL RequestingAssembly
...
  at  System.Runtime.Loader.AssemblyLoadContext.InvokeResolveEvent(ResolveEventHandler eventHandler, RuntimeAssembly ass
embly, String name)
   at System.Runtime.Loader.AssemblyLoadContext.OnAssemblyResolve(RuntimeAssembly assembly, String assemblyFullName)
   at System.Reflection.RuntimeAssembly.GetType(RuntimeAssembly assembly, String name, Boolean throwOnError, Boolean ign
oreCase, ObjectHandleOnStack type, ObjectHandleOnStack keepAlive)
   at System.Reflection.RuntimeAssembly.GetType(RuntimeAssembly assembly, String name, Boolean throwOnError, Boolean ign
oreCase, ObjectHandleOnStack type, ObjectHandleOnStack keepAlive)
   at System.Reflection.RuntimeAssembly.GetType(String name, Boolean throwOnError, Boolean ignoreCase)
   at System.Reflection.Assembly.GetType(String name)
```

Initial assessment looks like this must be happening when `Assembly.GetType(string name)` is using a generic type.

**Comment by sdmaclea** [comment]
I will try to create a repro.

**Comment by sdmaclea** [comment]
This is a repro for Microsoft.NETCore.App 2.2.4
```C#
using System;
using System.Reflection;

namespace repro
{
    class AGenericClass<T>
    {
    }

    class Program
    {
        static void Main(string[] args)
        {
            ResolveEventHandler handler = (sender, e) =>
            {
                if (e.RequestingAssembly == null)
                {
                        Console.WriteLine("e.RequestingAssembly == null");
                }
                return null;
            };

            AppDomain.CurrentDomain.AssemblyResolve += handler;

            typeof(Program).Assembly.GetType("repro.AGenericClass`1[[Bogus, BogusAssembly]]", false);
        }
    }
}
```

**Comment by sdmaclea** [comment]
Oddly enough the repro above does not fail in the current tip.

**Comment by sdmaclea** [comment]
It is also passing on the official .NET Core preview5 release.  
It is also failing on Microsoft.NETCore.App 2.15.

**Comment by sdmaclea** [comment]
I would assume the basic repro was fixed as part of #22213.  I think this means the customers issue will be resolved if they use official bits.

I think I can still cause this to break though....

**Comment by sdmaclea** [comment]
Closing as duplicate of #22213.
Feel free to reopen if my analysis is in error.
